components/feature/recentlyclosed/src/main/java/mozilla/components/feature/recentlyclosed/ext/ClosedTab.kt

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.recentlyclosed.ext
+
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.browser.state.action.RecentlyClosedAction
+import mozilla.components.browser.state.state.ClosedTab
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.browser.state.store.BrowserStore
+
+/**
+ * Restores the given [ClosedTab]. On restore, will invoke [onTabRestored] and remove the
+ * closed tab from the [BrowserState].
+ */
+fun ClosedTab.restoreTab(
+    browserStore: BrowserStore,
+    sessionManager: SessionManager,
+    onTabRestored: () -> Unit
+) {
+    val session = Session(this.url, false, SessionState.Source.RESTORED)
+    sessionManager.add(
```

Comment:
@pocmo from playing around with it, adding a session with my serialized `engineSessionState` was the only way to restore the tab fully? Or am I missing something? 🤔

Reply:
Yes, that is correct! :)